plugins/importer/src/main/java/com/managemc/importer/service/PunishmentImporterService.java

```diff
@@ -116,6 +112,24 @@ public void importMaxBans(CommandSender sender) {
     );
   }
 
+  public void importEssentialsX(CommandSender sender, File dataFolder) {
+    EssentialsPunishmentsCache punishmentsCache = new EssentialsPunishmentsCache();
+    PunishmentImporter importer = newPunishmentImporter(punishmentsCache.getBaseCache());
+    VanillaBanReader banReader = new VanillaBanReader();
```

Comment:
You will probably not use the Vanilla readers here, again - unless your ban plugin borrows Vanilla functionality, like Essentials does.